src/Service/EnvironmentService.php

```diff
@@ -63,6 +65,12 @@ public function __construct(
         $this->logger = $logger;
         $this->elasticaService = $elasticaService;
         $this->instanceId = $instanceId;
+
+        $environmentRepository = $doctrine->getRepository(Environment::class);
+        if (!$environmentRepository instanceof EnvironmentRepository) {
+            throw new \RuntimeException('Not found repository');
+        }
+        $this->environmentRepository = $environmentRepository;
```

Comment:
I don't like code in service constructors. It may impact the overall performances as they are called on each request. Personally, I would prefer a getter with a singleton. Or an autowiring stuff

```
        private function getEnvironmentRepository(): EnvironmentRepository
        {
                if (null !== $this->environmentRepository) {
                    return $this->environmentRepository;
                }
                $environmentRepository = $this->doctrine->getRepository(Environment::class);
                if (!$environmentRepository instanceof EnvironmentRepository) {
                    throw new \RuntimeException('Not found repository');
                }
                $this->environmentRepository = $environmentRepository;
                return $this->environmentRepository;
        }
```

Reply:
Should be fixed if we make a service for the Environment repository, but out of scope for this pr